pkg/collect/copy.go

```diff
@@ -60,8 +63,9 @@ func copyFiles(c *Collector, client *kubernetes.Clientset, pod corev1.Pod, copyC
 	if copyCollector.ContainerName != "" {
 		container = copyCollector.ContainerName
 	}
-
-	command := []string{"cat", copyCollector.ContainerPath}
+	//Command cd into the path directory, tars the target file or folder into a temp file, cat-copy it and removes the temp file.
+	//Not using a temp file may end up in error "Refusing to write archive contents to terminal"
+	command := []string{"sh", "-c", fmt.Sprintf("tar -C %v  -cf tmp.tar %v; cat tmp.tar; rm tmp.tar", path.Dir(copyCollector.ContainerPath), path.Base(copyCollector.ContainerPath))}
```

Comment:
what if the directory is not writeable? when does it end up in the error? how does this work?
https://github.com/kubernetes/kubernetes/blob/382107e6c84374b229e6188207ef026621286aa2/staging/src/k8s.io/kubectl/pkg/cmd/cp/cp.go#L315

Reply:
If you're really worried about the 'will refuse to write to terminal' bits, just pipe it to `| cat` - 'cat' doesn't care about being a binary file, and tar just sees that it's not going to the terminal.

So `tar -cf %s | cat`.

(Incidentally, you should probably use `%s` and not `%v` - that way we'll see improper type coercions)

Reply:
@laverya That actually did the trick! 👏 Thank you for the observation about the format verbs, it's a good point/
@emosbaugh For what I've read, it depends on the version of tar. Some of them consider printing the file to stdout a security risk and prevent it from happening.